etrago/cluster/networkclustering.py

```diff
@@ -781,23 +781,81 @@ def fetch():
         busmap = fetch()
 
     return busmap
+    
+def delete_ehv_buses_no_lines(network):
+    """
+    When there are AC buses totally isolated, this function deletes them in order
+    to make possible the creation of busmaps based on electrical connections
+    and other purposes. Additionally, it throws a warning to inform the user
+    in case that any correction should be done.
 
+    Parameters
+    ----------
+    network : pypsa.network
 
+    Returns
+    -------
+    None
+    """
+    lines = network.lines
+    buses_ac = network.buses[(network.buses.carrier == "AC") &
+                             (network.buses.country == "DE")]
+    buses_in_lines = set(list(lines.bus0) + list(lines.bus1))
+    buses_ac["with_line"] = buses_ac.index.isin(buses_in_lines)
+    buses_ac["with_load"] = buses_ac.index.isin(network.loads.bus)
+    buses_in_links = list(network.links.bus0)+list(network.links.bus1)
+    buses_ac["with_link"] = buses_ac.index.isin(buses_in_links)
+    buses_ac["with_gen"] = buses_ac.index.isin(network.generators.bus)
+
+    delete_buses = buses_ac[(buses_ac["with_line"] == False) &
+                            (buses_ac["with_load"] == False) &
+                            (buses_ac["with_link"] == False) &
+                            (buses_ac["with_gen"] == False)].index
+
+    if len(delete_buses):
+        logger.info(f"""
+                    
+                    ----------------------- WARNING ---------------------------
+                    THE FOLLOWING BUSES WERE DELETED BECAUSE THEY WERE ISOLATED:
+                        {delete_buses.to_list()}.
+                    IT IS POTENTIALLY A SIGN OF A PROBLEM IN THE DATASET
+                    ----------------------- WARNING ---------------------------
+                    
+                    """)
+                    
+    network.buses = network.buses.drop(delete_buses)
+    
+    delete_trafo = network.transformers[
+        (network.transformers.bus0.isin(delete_buses)) |
+        (network.transformers.bus1.isin(delete_buses))].index
+    
+    network.transformers = network.transformers.drop(delete_trafo)
+    
+    delete_sto_units = network.storage_units[
+        network.storage_units.bus.isin(delete_buses)].index
+    network.storage_units = network.storage_units.drop(delete_sto_units)
+    
+    return
+    
+    
 def ehv_clustering(self):
 
     if self.args["network_clustering_ehv"]:
 
         logger.info("Start ehv clustering")
 
         self.network.generators.control = "PV"
+
+        delete_ehv_buses_no_lines(self.network)        
+
         busmap = busmap_from_psql(self)
 
         self.network, busmap = cluster_on_extra_high_voltage(
-            self.network, busmap, with_time=True
+            self, busmap, with_time=True
         )
 
         self.update_busmap(busmap)
-
+        buses_by_country(self)
```

Comment:
I think this should be `self.buses_by_country()` , right?

Reply:
buses_by_country() was aggregated to the eTraGo methods.